<h4 align="center" style="height:1123px;font-family: 'Times New Roman'"><b>

    LAPORAN PRAKTIKUM

    ALGORITMA & PEMOGRAMAN

    “OPERATOR PADA PYTHON”

    Disusun Oleh :

    Alif Maulana Kesi

    2611533028

    Dosen Pengampu:

    Dr. Ir Wahyudi, S.T, M.T

    Asisten Praktikum:

    Sherly Sukmadira

<br>

 <img src="logo_unand.png" width="200">

<br>

    FAKULTAS TEKNOLOGI INFORMASI

    DEPARTEMEN INFORMATIKA

    UNIVERSITAS ANDALAS

    2026

</b></h4>
</div>

---
<div style="height:1123px;font-family: 'Times New Roman', Times, serif;font-size: 18px;">

<h2 align="center">KATA PENGANTAR</h2>
Puji syukur kehadirat Tuhan Yang Maha Esa karena atas rahmat dan karunia-Nya laporan praktikum yang berjudul “Operasi pada Python” ini dapat diselesaikan dengan baik. Laporan ini disusun untuk memenuhi salah satu tugas praktikum serta membantu memahami penggunaan berbagai operasi dalam bahasa pemrograman Python.

Dalam praktikum ini, penulis mempelajari beberapa jenis operasi yang terdapat pada Python, seperti operasi aritmatika, perbandingan, logika, penugasan, serta operasi lainnya yang dapat digunakan dalam pembuatan program. Dengan adanya praktikum ini, diharapkan penulis dapat memahami cara kerja operator dan menerapkannya dalam program sederhana.

Penulis menyadari bahwa laporan ini masih memiliki kekurangan. Oleh karena itu, kritik dan saran yang membangun sangat diharapkan agar laporan berikutnya dapat menjadi lebih baik. Semoga laporan ini dapat memberikan manfaat bagi penulis maupun pembaca dalam mempelajari dasar-dasar pemrograman Python.


<div align="right" style="margin-left:-30px;">
Padang, 2026
<br>
<br>
<br>
<br>
<br>
Alif Maulana Kesi E
<br>
2611533028
</div>
</div>

---
<div style="height:1123px;font-family: 'Times New Roman', Times, serif;font-size: 18px;">
<h1 align="center">BAB I<br>PENDAHULUAN</h1>

## 1.1 Latar Belakang
Python dikenal sebagai salah satu bahasa pemrograman yang banyak digunakan oleh pemula karena sintaksnya yang ringkas dan mudah dibaca. Meski terlihat sederhana, di balik setiap baris kode terdapat operator-operator yang bekerja untuk memproses data, membandingkan nilai, hingga mengatur alur logika program. Tanpa memahami operator ini, seorang pemrogram akan kesulitan menerjemahkan suatu masalah ke dalam bentuk instruksi yang benar.

Setiap jenis operator memiliki peran yang spesifik dalam sebuah program. Operator aritmatika berkaitan dengan perhitungan angka, operator perbandingan dan logika menjadi dasar pengambilan keputusan, sementara operator bitwise bekerja lebih dalam pada level representasi biner suatu data. Ketepatan dalam memilih dan menggunakan operator akan sangat memengaruhi hasil serta efisiensi program yang dibuat.

Berdasarkan hal tersebut, praktikum ini dirancang agar mahasiswa tidak hanya mengenal operator-operator dalam Python secara teori, tetapi juga terbiasa mempraktikkannya langsung melalui percobaan kode, sehingga pemahaman yang diperoleh lebih melekat dan siap diterapkan pada studi kasus pemrograman berikutnya.


## 1.2 Tujuan
1.  Memahami konsep dan jenis-jenis operator yang tersedia dalam bahasa pemrograman Python.
2. Menjelaskan cara kerja masing-masing operator beserta perbedaan fungsinya.
3. Mengimplementasikan operator-operator tersebut ke dalam potongan kode Python.
4. Menganalisis output yang dihasilkan dari setiap percobaan operator yang dijalankan.

## 1.3 Manfaat
1. Mahasiswa terbiasa membaca dan menulis sintaks Python dengan lebih percaya diri.
2. Praktikum ini membekali kemampuan logika dasar sebelum masuk ke struktur data yang lebih rumit.
3. Kesalahan umum dalam penggunaan operator dapat dikenali dan dihindari sejak awal.
4. Hasil praktikum ini menjadi bekal untuk mata kuliah pemrograman lanjutan.
</div>

---
<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">
<h1 align="center">BAB II<br>PEMBAHASAN</h1>

## 2.1 Dasar Teori
Python memiliki berbagai jenis operator yang digunakan untuk melakukan proses tertentu terhadap data dan variabel. Operator merupakan simbol atau kata khusus yang digunakan untuk melakukan operasi, seperti perhitungan, perbandingan, pemberian nilai, maupun pengolahan kondisi. Penggunaan operator menjadi salah satu dasar penting dalam pemrograman karena hampir setiap program membutuhkan operasi terhadap data.


2.1 Operator Aritmatika
Operator aritmatika berfungsi untuk mengolah nilai numerik dalam suatu perhitungan. Python mendukung operasi penjumlahan (+), pengurangan (-), perkalian (*), pembagian biasa (/), pembagian yang hasilnya dibulatkan ke bawah (//), sisa hasil bagi (%), serta pemangkatan (**). Urutan pengerjaannya mengikuti aturan prioritas matematika pada umumnya.

 2.2 Operator Perbandingan
Operator ini dipakai ketika dua nilai perlu dibandingkan satu sama lain, dan hasilnya selalu berupa nilai boolean, yaitu True atau False. Termasuk di dalamnya adalah operator kesetaraan (==, !=) dan operator relasional (>, <, >=, <=). Operator jenis ini banyak dipakai dalam pembuatan struktur percabangan seperti if-else.

 2.3 Operator Logika
Ketika sebuah program perlu mengecek lebih dari satu syarat sekaligus, operator logika berperan penting. and menghasilkan True hanya jika seluruh kondisi terpenuhi, or cukup satu kondisi saja yang benar, sedangkan not berfungsi membalikkan nilai kebenaran suatu ekspresi.

 2.4 Operator Penugasan
Operator penugasan digunakan untuk menyimpan atau memperbarui nilai suatu variabel. Selain tanda sama dengan (=) sebagai bentuk dasar, Python juga menyediakan operator gabungan seperti +=, -=, *=, dan /= yang memungkinkan penulisan kode menjadi lebih ringkas.

 2.5 Operator Bitwise
Berbeda dari operator lain yang bekerja pada nilai secara utuh, operator bitwise memproses data pada tingkat bit biner. Operator ini mencakup AND (&), OR (|), XOR (^), NOT (~), serta pergeseran bit ke kiri (<<) dan ke kanan (>>). Operator bitwise umumnya dipakai dalam pemrograman tingkat rendah atau optimisasi performa.

 2.6 Operator Identitas dan Keanggotaan
Operator identitas, yaitu is dan is not, digunakan untuk mengetahui apakah dua variabel menunjuk pada objek yang persis sama di dalam memori, bukan sekadar memiliki nilai yang sama. Sementara itu, operator keanggotaan in dan not in dipakai untuk memeriksa apakah suatu elemen terdapat di dalam struktur data seperti list, tuple, atau string.

## 2.2 Kode Program

### 2.2.1 Program aritmatika_2611531021
</div>

In [20]:
# Buat file dengan nama aritmatika_NIM.py
# Buat program untuk operator aritmatika dalam Python
# Nama variabel ditambah 4 digit nim terakhir contoh: angka1_1234
# Program ini menngunakan fungsi input()
# Nilai yang dimasukkan akan dikonversi menjadi tipe data integer

angka1_3028 = int(input("Input angka-1: "))
angka2_3028 = int(input("Input angka-2: "))

# Penjumlahan
hasil_3028 = angka1_3028 + angka2_3028
print("\nOperator Penjumlahan")
print("Hasil =",hasil_3028)

# Pengurangan
hasil_3028 = angka1_3028 - angka2_3028
print("\nOperator Pengurangan")
print("Hasil =",hasil_3028)

# Perkalian
hasil_3028 = angka1_3028 * angka2_3028
print("\nOperator Perkalian")
print("Hasil =",hasil_3028)

# Pembagian, pembagian bulat, dan sisa bagi
if angka2_3028 != 0:
    hasil_3028 = angka1_3028 / angka2_3028
    print("\nOperator Pembagian")
    print("Hasil =",hasil_3028)

    hasil_3028 = angka1_3028 // angka2_3028
    print("\nOperator Pembagian Bulat")
    print("Hasil =",hasil_3028)

    hasil_1021 = angka1_3028 % angka2_3028
    print("\nOperator Sisa Bagi")
    print("Hasil =",hasil_3028)
else:
    print("Angka kedua tidak boleh bernilai 0.")

# Pangkat
hasil_3028 = angka1_3028 ** angka2_3028
print("\nOperator Pangkat")
print("Hasil =",hasil_3028)


Operator Penjumlahan
Hasil = 30

Operator Pengurangan
Hasil = -10

Operator Perkalian
Hasil = 200

Operator Pembagian
Hasil = 0.5

Operator Pembagian Bulat
Hasil = 0

Operator Sisa Bagi
Hasil = 0

Operator Pangkat
Hasil = 100000000000000000000


<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

Penjelasan:
1. Membuat input angka<br>
angka1_3028 = int(input("Input angka-1: "))<br>
angka2_3028 = int(input("Input angka-2: "))<br>
Program meminta pengguna memasukkan dua angka. Fungsi input() digunakan untuk menerima data, sedangkan int() mengubah data yang dimasukkan menjadi tipe integer.
2. Operator Penjumlahan<br>
hasil_3028 = angka1_3028 + angka2_3028<br>
Operator + digunakan untuk menjumlahkan angka1_3028 dengan angka2_3028, kemudian hasilnya disimpan ke variabel hasil_3028.
3. Operator Pengurangan<br>
hasil_3028 = angka1_3028 - angka2_3028<br>
Operator - digunakan untuk mengurangi nilai angka1_3028 dengan angka2_3028.
4. Operator Perkalian<br>
hasil_3028 = angka1_3028 * angka2_3028<br>
Operator * digunakan untuk mengalikan kedua angka yang telah dimasukkan.
5. Operator Pembagian<br>
hasil_3028 = angka1_3028 / angka2_3028<br>
Operator / digunakan untuk melakukan pembagian. Hasil pembagian biasanya berupa tipe data float.
6. Operator Pembagian Bulat<br>
hasil_3028 = angka1_3028 // angka2_3028<br>
Operator // digunakan untuk melakukan pembagian dan mengambil hasil bilangan bulatnya saja, tanpa bagian desimal.
7. Operator Sisa Bagi<br>
hasil_3028 = angka1_3028 % angka2_3028<br>
Operator % digunakan untuk mendapatkan sisa dari hasil pembagian antara angka1_3028 dan angka2_3028.
8. Pengecekan angka kedua<br>
if angka2_3028 != 0:<br>
Program memeriksa apakah angka kedua tidak bernilai 0. Hal ini dilakukan karena pembagian dengan angka 0 tidak diperbolehkan.<br>
Jika angka2_3028 bernilai 0, program menjalankan:<br>
print("Angka kedua tidak boleh bernilai 0.")
9. Operator Pangkat<br>
hasil_3028 = angka1_3028 ** angka2_3028<br>
Operator ** digunakan untuk menghitung pangkat, yaitu angka1_3028 dipangkatkan dengan angka2_3028.


### 2.2.2 Program assigment_2611531021
</div>

In [21]:
# Buat file dengan nama assignment_2611533028
# Nama variabel ditambah 4 digit nim terakhir contoh: angka1_3028
# Program ini menggunakan fungsi input()
# Nilai yang dimasukkan akan dikonversi menjadi tipe data integer
# Program operator assignment dalam Python

angka1_3028 = int(input("Input angka-1: "))
angka2_3028 = int(input("Input angka-2: "))

print("\nNilai awal angka1_3028 =", angka1_3028 )
print("Nilai angka2_3028 =", angka2_3028 )

# Assignment biasa
hasil_3028 = angka1_3028
print("\nAssignment biasa (=)")
print("Hasil =", hasil_3028)

# Assignment penambahan 
hasil_3028 = angka1_3028
hasil_3028 += angka2_3028
print("\nAssignment penambahan (+=)")
print("Hasil =", hasil_3028)

# Assignment pengurangan
hasil_3028 = angka1_3028
hasil_3028 -= angka2_3028
print("\nAssignment penambahan (-=)")
print("Hasil =", hasil_3028)

# Assignment perkalian
hasil_3028 = angka1_3028
hasil_3028 *= angka2_3028
print("\nAssignment penambahan (*=)")
print("Hasil =", hasil_3028)

# Assignment pembagian, pembagian bulat, dan sisa bagi
if angka2_3028 != 0:
    hasil_3028 = angka1_3028
    hasil_3028 /= angka2_3028
    print("\nAssignment pembagian (/=)")
    print("Hasil =", hasil_3028)
    # Operator tambahan
    hasil_3028 = angka1_3028
    hasil_3028 //= angka2_3028
    print("\nAssignment pembagian bulat (//=)")
    print("Hasil =", hasil_3028)
    hasil_3028 = angka1_3028
    hasil_3028 %= angka2_3028
    print("\nAssignment sisa bagi (%=)")
    print("Hasil =", hasil_3028)
else:
    print("\nPembagian tidak dapat dilakukan")
    print("Angka kedua tidak boleh bernilai 0. ")

# Operator tambahan: assignment perpangkatan
hasil_3028 = angka1_3028
hasil_3028 **= angka2_3028
print("\nAssignment perpangkatan (**=)")
print("Hasil =", hasil_3028)


Nilai awal angka1_3028 = 10
Nilai angka2_3028 = 20

Assignment biasa (=)
Hasil = 10

Assignment penambahan (+=)
Hasil = 30

Assignment penambahan (-=)
Hasil = -10

Assignment penambahan (*=)
Hasil = 200

Assignment pembagian (/=)
Hasil = 0.5

Assignment pembagian bulat (//=)
Hasil = 0

Assignment sisa bagi (%=)
Hasil = 10

Assignment perpangkatan (**=)
Hasil = 100000000000000000000


<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

Penjelasan:
1. Membuat input angka<br>
angka1_3028 = int(input("Input angka-1: "))<br>
angka2_3028 = int(input("Input angka-2: "))<br>
Program meminta pengguna memasukkan dua angka. Fungsi input() digunakan untuk menerima masukan, sedangkan int() mengubah masukan tersebut menjadi tipe data integer.
2. Menampilkan nilai awal<br>
print("\nNilai awal angka1 =", angka1_3028)<br>
print("Nilai angka2 =", angka2_1021)<br>
Bagian ini digunakan untuk menampilkan nilai awal dari angka1_3028 dan angka2_3028.
3. Assignment biasa (=)<br>
hasil_3028 = angka1_3028<br>
Operator = digunakan untuk memberikan atau menetapkan nilai dari angka1_3028 ke variabel hasil_3028.
4. Assignment penambahan (+=)<br>
hasil_3028 = angka1_3028<br>
hasil_3028 += angka2_3028<br>
Operator += digunakan untuk menambahkan nilai angka2_3028 ke hasil_3028.<br>
Bentuk sederhananya:<br>
hasil_3028 = hasil_3028 + angka2_1021
5. Assignment pengurangan (-=)<br>
hasil_3028 = angka1_3028<br>
hasil_3028 -= angka2_3028<br>
Operator -= digunakan untuk mengurangi hasil_3028 dengan nilai angka2_3028.<br>
Bentuk sederhananya:<br>
hasil_3028 = hasil_3028 - angka2_3028
6. Assignment perkalian (*=)<br>
hasil_3028 = angka1_3028<br>
hasil_3028 *= angka2_3028<br>
Operator *= digunakan untuk mengalikan hasil_3028 dengan angka2_3028.<br>
Bentuk sederhananya:3028
hasil_3028 = hasil_3028 * angka2_3028
7. Assignment pembagian (/=)<br>
hasil_3028 = angka1_3028<br>
hasil_3028 /= angka2_3028<br>
Operator /= digunakan untuk membagi nilai hasil_3028 dengan angka2_3028.<br>
Bentuk sederhananya:<br>
hasil_3028 = hasil_3028 / angka2_3028<br>
Hasil pembagian menggunakan / biasanya berupa float.
8. Assignment pembagian bulat (//=)<br>
hasil_3028 = angka1_3028<br>
hasil_3028 //= angka2_3028<br>
Operator //= digunakan untuk melakukan pembagian bulat.<br>
Bentuk sederhananya:<br>
hasil_3028 = hasil_3028 // angka2_3028
9. Assignment sisa bagi (%=)<br>
hasil_3028 = angka1_3028<br>
hasil_3028 %= angka2_3028<br>
Operator %= digunakan untuk mendapatkan sisa hasil pembagian.<br>
Bentuk sederhananya:<br>
hasil_3028 = hasil_3028 % angka2_3028
10. Pengecekan nilai 0<br>
if angka2_3028 != 0:<br>
Program memeriksa apakah angka2_1021 tidak sama dengan 0. Pemeriksaan ini diperlukan karena operasi pembagian, pembagian bulat, dan sisa bagi tidak dapat dilakukan dengan pembagi 0.<br>
Jika nilainya 0, program menampilkan:<br>
print("\nPembagian tidak dapat dilakukan.")<br>
print("Angka kedua tidak boleh bernilai 0.")
11. Assignment perpangkatan (**=)<br>
hasil_3028 = angka1_3028<br>
hasil_3028 **= angka2_3028<br>
Operator **= digunakan untuk melakukan perpangkatan.<br>
Bentuk sederhananya:<br>
hasil_3028 = hasil_3028 ** angka2_3028
### 2.2.3 Program perbandingan_2611531021
</div>

In [22]:
# Buat file dengan nama perbandingan_2611533028.py
# Nama variabel ditambah 4 digit nim terakhir contoh: angka1_3028
# Progaram ini menggunakan fungsi input()
# Nilai yang dimasukkan akan dikonversi menjadi tipe data integer
# Program operator perbandingan dalam Python

angka1_3028 = int(input("input angka-1:"))
angka2_3028 = int(input("input angka-2:"))

# Lebih besar dari
hasil_3028 = angka1_3028 > angka2_3028
print("\nOperator lebih besar dari")
print("angka1_3028 > angka2_3028 =", hasil_3028)

# Lebih kecil dari
hasil_3028 = angka1_3028 < angka2_3028
print("\nOperator lebih kecil dari")
print("angka1_3028 < angka2_3028 =", hasil_3028)

# Lebih kecil dari atau sama dengan
hasil_3028 = angka1_3028 <= angka2_3028
print("\nOperator lebih kecil atau sama dengan")
print("angka1_3028 <= angka2_3028 =", hasil_3028)

# Sama dengan
hasil_3028 = angka1_3028 = angka2_3028
print("\nOperator sama dengan")
print("angka1_3028 = angka2_3028 =", hasil_3028)

# Tidak sama dengan 
hasil_3028 = angka1_3028 = angka2_3028
print("\nOperator tidak sama dengan")
print("angka1_3028 != angka2_3028 =", hasil_3028)

# Tambahan: perbandingan berantai dalam Python
hasil_3028 = 0 < angka1_3028 < 100
print("\nPerbandingan berantai")
print("0 < angka1_3028 < 100 =", hasil_3028)

hasil_3028 = 0 < angka2_3028 < 100
print("\nPerbandingan berantai")
print("0 < angka2_3028 < 100 =", hasil_3028)


Operator lebih besar dari
angka1_3028 > angka2_3028 = False

Operator lebih kecil dari
angka1_3028 < angka2_3028 = True

Operator lebih kecil atau sama dengan
angka1_3028 <= angka2_3028 = True

Operator sama dengan
angka1_3028 = angka2_3028 = 20

Operator tidak sama dengan
angka1_3028 != angka2_3028 = 20

Perbandingan berantai
0 < angka1_3028 < 100 = True

Perbandingan berantai
0 < angka2_3028 < 100 = True


<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

Penjelasan:
1. Membuat input angka<br>
angka1_3028 = int(input("Input angka-1: "))<br>
angka2_3028 = int(input("Input angka-2: "))<br>
Program meminta pengguna memasukkan dua angka. Fungsi input() digunakan untuk menerima masukan, sedangkan int() mengubah masukan menjadi tipe data integer.
2. Operator lebih besar dari (>)<br>
hasil_3028 = angka1_3028 > angka2_1021<br>
Operator > digunakan untuk mengecek apakah angka1_3028 lebih besar daripada angka2_3028.<br>
Hasilnya berupa nilai Boolean, yaitu True atau False.
3. Operator lebih kecil dari (<)<br>
hasil_3028 = angka1_3028 < angka2_1021<br>
Operator < digunakan untuk mengecek apakah angka1_1021 lebih kecil daripada angka2_3028.
4. Operator lebih besar dari atau sama dengan (>=)<br>
hasil_3028 = angka1_3028 >= angka2_3028<br>
Operator >= mengecek apakah angka1_3028 lebih besar atau sama dengan angka2_3028.
5. Operator lebih kecil dari atau sama dengan (<=)<br>
hasil_3028 = angka1_3028 <= angka2_3028<br>
Operator <= mengecek apakah angka1_3028 lebih kecil atau sama dengan angka2_3028.
6. Operator sama dengan (==)<br>
hasil_3028 = angka1_3028 == angka2_3028<br>
Operator == digunakan untuk mengecek apakah kedua angka memiliki nilai yang sama.<br>
Perlu dibedakan antara = dan ==.<br>
= digunakan untuk memberikan nilai, sedangkan == digunakan untuk membandingkan nilai.
7. Operator tidak sama dengan (!=)<br>
hasil_3028 = angka1_3028 != angka2_3028<br>
Operator != digunakan untuk mengecek apakah angka1_3028 dan angka2_3028 memiliki nilai yang berbeda.
8. Perbandingan berantai untuk angka pertama<br>
hasil_1021 = 0 < angka1_3028 < 100<br>
Kode tersebut merupakan perbandingan berantai (chained comparison). Artinya, program mengecek apakah angka1_3028 berada di antara 0 dan 100, dengan syarat:<br>
0 < angka1_3028 dan angka1_3028 < 100<br>
Jika kedua kondisi benar, hasilnya True.
9. Perbandingan berantai untuk angka kedua<br>
hasil_3028 = 0 < angka2_3028 < 100<br>
Bagian ini memiliki fungsi yang sama, tetapi digunakan untuk mengecek apakah angka2_1021 berada di antara 0 dan 100.

### 2.2.4 Program logika_2611531021
</div>

In [23]:
# Buat file dengan nama logika_2611533028.py
# Nama variabel ditambah 4 digit nim terakhir contoh: a1_3028
# Progaram ini menggunakan fungsi input()
# Program operator logika dalam Python

# Memasukkan nilai Boolean
# Input tidak peka terhadap huruf besar dan kecil
a1_3028 =  input("Inout nilai boolean-1 (true/false): ").strip().lower() == "true"
a2_3028 =  input("Inout nilai boolean-1 (true/false): ").strip().lower() == "true"

print("\nA1 =", a1_3028)
print("\nA2 =", a2_3028)

# Konjungsi: bernilai True jika keduanya True
hasil_3028 = a1_3028 and a2_3028
print("\nKonjungsi (AND)")
print("A1 and A2 =", hasil_3028)

# Disjungsi: bernilai True jika keduanya True
hasil_3028 = a1_3028 or a2_3028
print("\nDisjungsi (OR)")
print("A1 OR A2 =", hasil_3028)

# Negasi A1: membalik nilai A1
hasil_3028 = not a1_3028
print("\nNegasi A1 (NOT)")
print("not A1 =", hasil_3028)

# Negasi A2: membalik nilai A2
hasil_3028 = not a2_3028
print("\nNegasi A2 (NOT)")
print("not A2 =", hasil_3028)

# XOR: bernilai True jika kedua nilai berbeda
hasil_3028 = a1_3028 != a2_3028
print("\nDisjungsi Eksklusif (XOR)")
print("A1 XOR A2 =", hasil_3028)



A1 = False

A2 = False

Konjungsi (AND)
A1 and A2 = False

Disjungsi (OR)
A1 OR A2 = False

Negasi A1 (NOT)
not A1 = True

Negasi A2 (NOT)
not A2 = True

Disjungsi Eksklusif (XOR)
A1 XOR A2 = False


<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px3028
Penjelasan:
1. Memasukkan nilai Boolean<br>
a1_3028 = input("Input nilai boolean-1 (true/false): ").strip().lower() == "true"<br>
a2_3028 = input("Input nilai boolean-2 (true/false): ").strip().lower() == "true"<br>
Program meminta pengguna memasukkan nilai true atau false.<br>
input() menerima masukan dari pengguna.<br>
.strip() menghilangkan spasi di awal dan akhir.<br>
.lower() mengubah huruf menjadi huruf kecil sehingga TRUE, True, dan true dapat diproses dengan cara yang sama.<br>
== "true" membandingkan masukan dengan teks "true" dan menghasilkan nilai Boolean True atau False.
2. Menampilkan nilai Boolean<br>
print("\nA1 = ", a1_3028)<br>
print("A2 = ", a2_3028)<br>
Bagian ini menampilkan nilai Boolean yang telah dimasukkan dan dikonversi oleh program.
3. Konjungsi (AND)
hasil_3028 = a1_3028 and a2_3028<br>
Operator and digunakan untuk menggabungkan dua kondisi. Hasilnya akan True hanya jika kedua nilai bernilai True.<br>
Contoh:<br>
A1 = True<br>
A2 = True<br>
Hasil = True<br>
Jika salah satu atau keduanya False, hasilnya False.
4. Disjungsi (OR)<br>
hasil_3028 = a1_3028 or a2_3028<br>
Operator or menghasilkan True jika minimal salah satu nilai bernilai True.<br>
Hasilnya hanya False jika kedua nilai bernilai False.
5. Negasi A1 (NOT)<br>
hasil_3028 = not a1_3028<br>
Operator not digunakan untuk membalik nilai Boolean dari a1_3028.<br>
Jika A1 = True, maka not A1 = False.<br>
Jika A1 = False, maka not A1 = True.
6. Negasi A2 (NOT)<br>
hasil_3028 = not a2_3028<br>
Bagian ini memiliki fungsi yang sama dengan sebelumnya, tetapi digunakan untuk membalik nilai Boolean a2_1021.
7. Disjungsi Eksklusif (XOR)<br>
hasil_3028 = a1_3028 != a2_3028<br>
Operator XOR digunakan untuk menghasilkan True jika kedua nilai berbeda.

### 2.2.5 Program lainnya_2611531021
</div>

In [24]:
# Buat file dengan nama lainnya_NIM.py
# Nama variabel ditambah 4 digit nim terakhir contoh: angka1_3028
# Program ini menggunakan fungsi input()
# Program operator keanggotaan dan  identitas

print("========================================")
print("1. OPERATOR KEANGGOTAAN")
print("========================================")

# Input beberapa data yang dipisahkan dengan koma
input_data_3028 = input("Masukkan beberapa angka, pisahkan dengan koma: ")

# Mengubah input menjadi list integer
data_3028 = [int(angka.strip()) for angka in input_data_3028.split(",")]

nilai_dicari_3028 = int(input("Masukkan angka yang ingin dicari: "))

# Operator in
hasil_3028 = nilai_dicari_3028 in data_3028
print("\nOperator keanggotaan IN")
print(nilai_dicari_3028, "in", data_3028, "=", hasil_3028)

# Operator not in
hasil_3028 = nilai_dicari_3028 not in data_3028
print("\nOperator keanggotaan NOT IN")
print(nilai_dicari_3028, "not in", data_3028, "=", hasil_3028)

print("\n========================================")
print("2. OPERATOR IDENTITAS")
print("========================================")

# objek1 menggunakan list dari input pengguna
objek1_3028 = data_3028

# objek2 merujuk pada objek yang sama dengan objek1
objek2_3028 = objek1_3028

# objek3 memiliki isi sama, tetapi merupakan objek baru
objek3_3028 = data_3028.copy()

print("objek1 =", objek1_3028)
print("objek2 =", objek2_3028)
print("objek3 =", objek3_3028)

# Operator is
hasil_3028 = objek1_3028 is objek2_3028
print("\nOperator identitas IS")
print("objek1_3028 is objek2_3028 =", hasil_3028)

# Operator is not
hasil_3028 = objek1_3028 is not objek3_3028
print("\nOperator identitas IS NOT")
print("objek1_3028 is not objek3_3028 =", hasil_3028)

# Membandingkan identitas dan nilai
print("\nPerbandingan identitas dan nilai")
print("objek1_3028 is objek3_3028 =", objek1_3028 is objek3_3028)
print("objek1_3028 == objek3_3028 =", objek1_3028 == objek3_3028)

1. OPERATOR KEANGGOTAAN

Operator keanggotaan IN
10 in [10, 20, 30, 40] = True

Operator keanggotaan NOT IN
10 not in [10, 20, 30, 40] = False

2. OPERATOR IDENTITAS
objek1 = [10, 20, 30, 40]
objek2 = [10, 20, 30, 40]
objek3 = [10, 20, 30, 40]

Operator identitas IS
objek1_3028 is objek2_3028 = True

Operator identitas IS NOT
objek1_3028 is not objek3_3028 = True

Perbandingan identitas dan nilai
objek1_3028 is objek3_3028 = False
objek1_3028 == objek3_3028 = True


<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

Penjelasan:
1. Menampilkan judul program<br>
print("======================================")<br>
print("1. OPERATOR KEANGGOTAAN")<br>
print("======================================")<br>
Digunakan untuk menampilkan judul bagian Operator Keanggotaan agar output program lebih terstruktur.
2. Menerima input beberapa angka
input_data_3028 = input("Masukkan beberapa angka, pisahkan dengan koma: ")<br>
Program meminta pengguna memasukkan beberapa angka yang dipisahkan dengan tanda koma. Input tersebut awalnya disimpan sebagai string.<br>
Contoh:<br>
10, 20, 30, 40
3. Mengubah input menjadi list integer<br>
data_3028 = [int(angka.strip()) for angka in input_data_3028.split(',')]<br>
Kode ini mengubah input string menjadi list yang berisi integer.<br>
.split(',') memisahkan data berdasarkan tanda koma.<br>
.strip() menghilangkan spasi yang tidak diperlukan.<br>
int() mengubah setiap data menjadi integer.<br>
[ ... for ... ] digunakan untuk membuat list.<br>
Contoh:<br>
Input: 10, 20, 30<br>
Hasil: [10, 20, 30]
4. Memasukkan nilai yang ingin dicari<br>
nilai_dicari_3028 = int(input("Masukkan angka yang ingin dicari: "))<br>
Program meminta pengguna memasukkan angka yang akan dicari di dalam data_3028. Input dikonversi menjadi tipe data integer.
5. Operator keanggotaan in<br>
hasil_3028 = nilai_dicari_3028 in data_3028<br>
Operator in digunakan untuk mengecek apakah suatu nilai terdapat di dalam sebuah list.<br>
Jika nilai ditemukan, hasilnya True. Jika tidak ditemukan, hasilnya False.<br>
Contoh:<br>
data = [10, 20, 30]<br>
20 in data → True<br>
50 in data → False
6. Operator keanggotaan not in<br>
hasil_3028 = nilai_dicari_3028 not in data_3028<br>
Operator not in digunakan untuk mengecek apakah suatu nilai tidak terdapat di dalam list.<br>
Jika nilai tidak ditemukan, hasilnya True. Jika ditemukan, hasilnya False.
7. Menampilkan judul operator identitas<br>
print("======================================")<br>
print("2. OPERATOR IDENTIAS")<br>
print("======================================")<br>
Bagian ini menampilkan judul untuk bagian Operator Identitas.
8. Membuat object1<br>
object1_3028 = data_3028<br>
Variabel object1_3028 menunjuk ke objek list yang sama dengan data_3028.
9. Membuat object2<br>
object2_3028 = object1_3028<br>
Variabel object2_3028 menunjuk ke objek yang sama dengan object1_3028.<br>
Artinya, object1_3028 dan object2_3028 memiliki identitas objek yang sama.
10. Membuat object3<br>
object3_3028 = data_3028.copy()<br>
copy() membuat objek list baru yang memiliki isi yang sama dengan data_3028.<br>
Jadi, nilai object3_1021 sama dengan object1_3028, tetapi objeknya berbeda.
11. Menampilkan isi objek<br>
print("object1 =",object1_3028)<br>
print("object2 =",object2_3028)<br>
print("object3 =",object3_3028)<br>
Digunakan untuk menampilkan isi dari ketiga objek.<br>
Karena ketiganya memiliki isi yang sama, hasil yang ditampilkan akan terlihat sama.
12. Operator identitas is<br>
hasil_3028 = object1_3028 is object2_3028<br>
Operator is digunakan untuk mengecek apakah dua variabel menunjuk ke objek yang sama.<br>
Karena object2_3028 dibuat dari object1_3028, hasilnya adalah:<br>
True
13. Operator identitas is not<br>
hasil_3028 = object1_3028 is not object2_3028<br>
Operator is not digunakan untuk mengecek apakah dua variabel menunjuk ke objek yang berbeda.<br>
Karena object1_3028 dan object2_3028 menunjuk objek yang sama, hasilnya adalah:<br>
False
14. Membandingkan identitas dan nilai<br>
print("objek1_3028 is objek3_3028 =",object1_3028 is object3_3028)<br>
print("objek1_3028 == objek3_3028 =",object1_3028 == object3_3028)<br>
Baris pertama menggunakan is untuk membandingkan identitas objek. Karena object3_3028 dibuat menggunakan .copy(), objeknya berbeda sehingga hasilnya:<br>
False<br>
Baris kedua menggunakan == untuk membandingkan nilai atau isi objek. Karena isi object1_3028 dan object3_3028 sama, hasilnya:<br>
True

### 2.2.6 Program bitwise_2611531021
</div>

In [25]:
# Buat file dengan nama bitwise_NIM.py
# Nama variabel ditambah 4 digit nim terakhir contoh: angka1_1234
# Program ini menggunakan fungsi input()

print("\n========================================")
print("3. OPERATOR BITWISE")
print("========================================")

angka1_3028 = int(input("Masukkan angka bitwise-1: "))
angka2_3028 = int(input("Masukkan angka bitwise-2: "))

print("\nAngka dalam bentuk desimal dan biner")
print("angka1_3028 =", angka1_3028, "| biner =", bin(angka1_3028))
print("angka2_3028 =", angka2_3028, "| biner =", bin(angka2_3028))

# Bitwise AND
hasil_3028 = angka1_3028 & angka2_3028
print("\nBitwise AND (&)")
print(angka1_3028, "&", angka2_3028, "=", hasil_3028)
print("Biner hasil =", bin(hasil_3028))
print("Biner hasil (8 bit) =", format(hasil_3028, "08b"))

# Bitwise OR
hasil_3028 = angka1_3028 | angka2_3028
print("\nBitwise OR (|)")
print(angka1_3028, "|", angka2_3028, "=", hasil_3028)
print("Biner hasil =", bin(hasil_3028))
print("Biner hasil (8 bit) =", format(hasil_3028, "08b"))

# Bitwise XOR
hasil_3028 = angka1_3028 ^ angka2_3028
print("\nBitwise XOR (^)")
print(angka1_3028, "^", angka2_3028, "=", hasil_3028)
print("Biner hasil =", bin(hasil_3028))
print("Biner hasil (8 bit) =", format(hasil_3028, "08b"))

# Bitwise NOT
hasil_3028 = ~angka1_3028
print("\nBitwise NOT (~)")
print("~", angka1_3028, "=", hasil_3028)
print("Biner hasil =", bin(hasil_3028))
print("Biner hasil (8 bit) =", format(hasil_3028, "08b"))

# Bitwise geser kiri
jumlah_geser_3028 = int(input("\nMasukkan jumlah pergeseran bit: "))

hasil_3028 = angka1_3028 << jumlah_geser_3028
print("\nBitwise geser kiri (<<)")
print(angka1_3028, "<<", jumlah_geser_3028, "=", hasil_3028)
print("Biner hasil =", bin(hasil_3028))
print("Biner hasil (8 bit) =", format(hasil_3028, "08b"))

# Bitwise geser kanan
hasil_3028 = angka1_3028 >> jumlah_geser_3028
print("\nBitwise geser kanan (>>)")
print(angka1_3028, ">>", jumlah_geser_3028, "=", hasil_3028)
print("Biner hasil =", bin(hasil_3028))
print("Biner hasil (8 bit) =", format(hasil_3028, "08b"))


3. OPERATOR BITWISE

Angka dalam bentuk desimal dan biner
angka1_3028 = 5 | biner = 0b101
angka2_3028 = 5 | biner = 0b101

Bitwise AND (&)
5 & 5 = 5
Biner hasil = 0b101
Biner hasil (8 bit) = 00000101

Bitwise OR (|)
5 | 5 = 5
Biner hasil = 0b101
Biner hasil (8 bit) = 00000101

Bitwise XOR (^)
5 ^ 5 = 0
Biner hasil = 0b0
Biner hasil (8 bit) = 00000000

Bitwise NOT (~)
~ 5 = -6
Biner hasil = -0b110
Biner hasil (8 bit) = -0000110

Bitwise geser kiri (<<)
5 << 1 = 10
Biner hasil = 0b1010
Biner hasil (8 bit) = 00001010

Bitwise geser kanan (>>)
5 >> 1 = 2
Biner hasil = 0b10
Biner hasil (8 bit) = 00000010


<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

Penjelasan:
1. Menampilkan judul<br>
print("======================================")<br>
print("3. OPERATOR BITWISE")<br>
print("======================================")<br>
Digunakan untuk menampilkan judul bagian program agar output terlihat lebih terstruktur.
2. Memasukkan dua angka<br>
angka1_3028 = int(input("Masukkan angka bitwise-1: "))<br>
angka2_3028 = int(input("Masukkan angka bitwise-2: "))<br>
Program meminta pengguna memasukkan dua angka. input() menerima masukan, sedangkan int() mengubah masukan menjadi tipe data integer.
3. Menampilkan angka desimal dan biner<br>
print("\nAngka dalam bentuk desimal dan biner")<br>
print("angka1_3028 =",angka1_3028,"| biner =",bin(angka1_3028))<br>
print("angka2_3028 =",angka2_3028,"| biner =",bin(angka2_3028))<br>
Menampilkan nilai angka dalam bentuk desimal dan biner. Fungsi bin() digunakan untuk mengubah bilangan integer menjadi bentuk biner.
4. Operasi AND (&)<br>
hasil_3028 = angka1_3028 & angka2_3028<br>
Operator & membandingkan setiap bit dari kedua angka. Hasil bit akan 1 jika kedua bit bernilai 1.<br>
print(angka1_3028,"&",angka2_3028,hasil_3028)<br>
print("Biner hasil =",bin(hasil_3028))<br>
print("Biner hasil (8 bit) =",format(hasil_3028,"08b"))<br>
Menampilkan hasil dalam bentuk desimal, biner biasa, dan biner dengan panjang 8 bit. format(..., "08b") digunakan untuk menampilkan biner minimal 8 digit.
5. Operasi OR (|)<br>
hasil_3028 = angka1_3028 | angka2_3028
Operator | menghasilkan bit 1 jika salah satu atau kedua bit bernilai 1.<br>
print(angka1_3028,"|",angka2_3028,hasil_3028)<br>
print("Biner hasil =",bin(hasil_3028))<br>
print("Biner hasil (8 bit) =",format(hasil_3028,"08b"))<br>
Menampilkan hasil operasi dalam bentuk desimal dan biner.
6. Operasi XOR (^)<br>
hasil_3028 = angka1_3028 ^ angka2_3028<br>
Operator ^ menghasilkan bit 1 jika kedua bit berbeda.<br>
Contoh:<br>
0 ^ 0 = 0<br>
0 ^ 1 = 1<br>
1 ^ 0 = 1<br>
1 ^ 1 = 0<br>
Bagian print() berikutnya menampilkan hasil dalam bentuk desimal dan biner.
7. Operasi NOT (~)<br>
hasil_3028 = ~angka1_3028<br>
Operator ~ digunakan untuk membalik bit dari angka_3028. Bit 1 menjadi 0 dan bit 0 menjadi 1.<br>
print("~",angka1_3028, "=",hasil_3028)<br>
Menampilkan hasil operasi NOT.<br>
8. Memasukkan jumlah pergeseran bit<br>
jumlah_geser_3028 = int(input("\nMasukkan jumlah pergeseran bit: "))<br>
Program meminta jumlah posisi bit yang ingin digeser. Nilai tersebut dikonversi menjadi integer.
9. Pergeseran bit ke kiri (<<)<br>
hasil_3028 = angka1_3028 << jumlah_geser_3028<br>
Operator << menggeser bit angka1_3028 ke sebelah kiri sebanyak jumlah_geser_3028 posisi.<br>
Secara sederhana, pergeseran kiri satu posisi biasanya sama dengan mengalikan bilangan dengan 2.<br>
Contoh:<br>
5 = 0101<br>
5 << 1 = 1010 = 10
10. Menampilkan hasil geser kiri<br>
print("\nBitwise geser kiri (<<)")<br>
print(angka1_3028,"<<",jumlah_geser_3028,"=",hasil_3028)<br>
print("Biner hasil =",bin(hasi_3028))<br>
print("Biner hasil (8 bit) =",format(hasil_3028,"08b"))<br>
Menampilkan hasil pergeseran dalam bentuk desimal, biner, dan biner 8 bit.
11. Pergeseran bit ke kanan (>>)<br>
hasil_3028 = angka1_3028 >> jumlah_geser_3028<br>
Operator >> menggeser bit angka1_3028 ke sebelah kanan sebanyak jumlah posisi yang ditentukan.<br>
Secara sederhana, untuk bilangan positif, pergeseran kanan satu posisi biasanya sama dengan membagi bilangan dengan 2 dan mengambil bagian bulatnya.<br>
Contoh:<br>
10 = 1010<br>
10 >> 1 = 0101 = 5
12. Menampilkan hasil geser kanan<br>
print("\nBitwise geser kiri (>>)")<br>
print(angka1_3028,">>",jumlah_geser_3028,"=",hasil_3028)<br>
print("Biner hasil =",bin(hasil_3028))<br>
print("Biner hasil (8 bit) =",format(hasil_1021,"08b"))

---
## Tugas Praktikum

**Soal :**
1. File Program Python: Buatlah program Python yang menyelesaikan studi kasus terpadu dengan nama berkas tugas3_nim.py (ganti nim dengan nomor induk mahasiswa masing-masing).
2. Laporan Praktikum: Susunlah laporan praktikum pekan 3 yang mendokumentasikan seluruh latihan praktikum serta penjelasan rinci alur logika program tugas dengan nama berkas LapPrak_ALpro3.ipynb.
3. Penyimpanan di GitHub: Simpan kedua file tersebut ke dalam folder Pekan3 di dalam repositori praktikum masing-masing. Format repositori: 2026_PrakAlpro_[Kelas]_[NIM]. Lakukan tahapan staging, commit, dan push ke akun GitHub pribadi sebelum batas waktu pengumpulan berakhir.
4. Template Laporan: Format penulisan laporan wajib mengikuti panduan dan template yang telah disediakan pada praktikum.

**Kode Jawaban :**

a. Kode Program tugas3_2611533028
</div>

In [ ]:
print("=== SISTEM RESERVASI TIKET BIOSKOP ===")

# ------------------------------
# 1. Input data penonton
# ------------------------------
nama_3028 = input("Masukkan Nama Penonton : ")
status_member_3028 = input("Masukkan Status Keanggotaan (member/nonmember) : ")
total_harga_3028 = int(input("Masukkan Total Harga Tiket : "))
jumlah_tiket_3028 = int(input("Masukkan Jumlah Tiket : "))
kode_voucher_3028 = input("Masukkan Kode Voucher : ")

# ------------------------------
# 2. Menampilkan data reservasi
# ------------------------------
print("\n=== DATA RESERVASI ===")
print(f"Nama Penonton           : {nama_3028}")
print(f"Status Keanggotaan      : {status_member_3028}")
print(f"Total Harga Tiket       : Rp{total_harga_3028}")
print(f"Jumlah Tiket            : {jumlah_tiket_3028}")
print(f"Kode Voucher            : {kode_voucher_3028}")

# ------------------------------
# 3. Validasi syarat-syarat
# ------------------------------
daftar_voucher_3028 = ["NONTON10", "NONTON20", "HEMATWEEKEND", "SPESIALMHS"]  # daftar voucher valid

syarat_harga_3028 = total_harga_3028 >= 150000            # syarat minimal transaksi
syarat_jumlah_3028 = jumlah_tiket_3028 >= 2               # syarat minimal jumlah tiket
status_valid_3028 = status_member_3028 == "member"        # cek status member
voucher_valid_3028 = kode_voucher_3028 in daftar_voucher_3028  # cek voucher ada di daftar

print("\n=== HASIL VALIDASI ===")
print(f"Harga >= Rp150000       : {syarat_harga_3028}")
print(f"Jumlah Tiket >= 2       : {syarat_jumlah_3028}")
print(f"Status Member           : {status_valid_3028}")
print(f"Voucher Tersedia        : {voucher_valid_3028}")
print(f"Mendapatkan Diskon      : {syarat_harga_3028 or syarat_jumlah_3028}")
print(f"Mendapatkan Voucher     : {voucher_valid_3028}")

# ------------------------------
# 4. Perhitungan pembayaran
# ------------------------------
diskon = 0.10

print("\n=== HASIL PERHITUNGAN ===")
print(f"Diskon                  : Rp{diskon * total_harga_3028}")
print(f"Total Pembayaran        : Rp{total_harga_3028 - total_harga_3028 * diskon}")
print(f"Rata-rata Harga Tiket   : Rp{total_harga_3028 / jumlah_tiket_3028}")

# ==========================================
# 5. HAK AKSES PENONTON (encoding ke bit)
# ==========================================
print("\n=== HAK AKSES PENONTON ===")
print("Kode Hak Akses          : bit0=Member, bit1=Harga, bit2=Jumlah, bit3=Voucher")
print(f"Member Access           : {status_valid_3028}")
print(f"Voucher Access          : {voucher_valid_3028}")
print("Studio Premium Access   : ...")

# Menggabungkan 4 status boolean menjadi satu kode biner 4-bit
kode_reservasi_3028 = int(status_valid_3028) << 0 | int(syarat_harga_3028) << 1 | int(syarat_jumlah_3028) << 2 | int(voucher_valid_3028) << 3
kode_referensi_3028 = int(status_valid_3028) << 0 | int(syarat_harga_3028) << 1 | int(voucher_valid_3028) << 3

print("\n=== OPERASI BITWISE ===")
print("=== Kode Status Reservasi ===")
print(f"{format(int(status_valid_3028) << 0, '04b')} | {format(int(syarat_harga_3028) << 1, '04b')} | "
      f"{format(int(syarat_jumlah_3028) << 2, '04b')} | {format(int(voucher_valid_3028) << 3, '04b')}")
print(f"Kode Biner              : {format(kode_reservasi_3028, '04b')}")
print(f"Kode Desimal            : {kode_reservasi_3028}")

print("\n=== Pemeriksaan Status (AND) ===")
print("Cek Member")
print(f"{format(kode_reservasi_3028, '04b')} & {format(int(status_valid_3028) << 0, '04b')}")
print(f"Hasil Biner    : {format(kode_reservasi_3028 & (int(status_valid_3028) << 0), '04b')}")
print(f"Hasil Desimal  : {kode_reservasi_3028 & (int(status_valid_3028) << 0)}")

print("Cek Voucher")
print(f"{format(kode_reservasi_3028, '04b')} & {format(int(voucher_valid_3028) << 3, '04b')}")
print(f"Hasil Biner    : {format(kode_reservasi_3028 & (int(voucher_valid_3028) << 3), '04b')}")
print(f"Hasil Desimal  : {kode_reservasi_3028 & (int(voucher_valid_3028) << 3)}")

print("\n=== Perbandingan Status (XOR) ===")
print(f"Kode Reservasi : {format(kode_reservasi_3028, '04b')}")
print(f"Kode Referensi : {format(kode_referensi_3028, '04b')}")
print(f"{format(kode_reservasi_3028, '04b')} ^ {format(kode_referensi_3028, '04b')}")
print(f"Hasil Biner    : {format(kode_reservasi_3028 ^ kode_referensi_3028, '04b')}")
print(f"Hasil Desimal  : {kode_reservasi_3028 ^ kode_referensi_3028}")

print("\n=== Shift (Upgrade Level Akses) ===")
print(f"{format(kode_reservasi_3028, '04b')} << 1")
print(f"Hasil Biner    : {format(kode_reservasi_3028 << 1, '04b')}")
print(f"Hasil Desimal  : {kode_reservasi_3028 << 1}")

print("\n=== SELESAI ===")



=== SISTEM RESERVASI TIKET BIOSKOP ===



=== DATA RESERVASI ===
Nama Penonton           : Alif
Status Keanggotaan      : non member
Total Harga Tiket       : Rp100000
Jumlah Tiket            : 2
Kode Voucher            : NONTON10

=== HASIL VALIDASI ===
Harga >= Rp150000       : False
Jumlah Tiket >= 2       : True
Status Member           : False
Voucher Tersedia        : True
Mendapatkan Diskon      : True
Mendapatkan Voucher     : True

=== HASIL PERHITUNGAN ===
Diskon                  : Rp10000.0
Total Pembayaran        : Rp90000.0
Rata-rata Harga Tiket   : Rp50000.0

=== HAK AKSES PENONTON ===
Kode Hak Akses          : bit0=Member, bit1=Harga, bit2=Jumlah, bit3=Voucher
Member Access           : False
Voucher Access          : True
Studio Premium Access   : ...

=== OPERASI BITWISE ===
=== Kode Status Reservasi ===
0000 | 0000 | 0100 | 1000
Kode Biner              : 1100
Kode Desimal            : 12

=== Pemeriksaan Status (AND) ===
Cek Member
1100 & 0000
Hasil Biner    : 0000
Hasil Desimal  : 0
Cek Voucher
1100 & 1000
Ha

<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

**Penjelasan :** 
**Penjelasan :**

1. Menampilkan judul program
`print("=== SISTEM RESERVASI TIKET BIOSKOP ===")`
Menampilkan judul utama program.

2. Memasukkan data penonton
```
nama_3028 = input("Masukkan Nama Penonton : ")
status_member_3028 = input("Masukkan Status Keanggotaan (member/nonmember) : ")
total_harga_3028 = int(input("Masukkan Total Harga Tiket : "))
jumlah_tiket_3028 = int(input("Masukkan Jumlah Tiket : "))
kode_voucher_3028 = input("Masukkan Kode Voucher : ")
```
Program meminta nama, status keanggotaan, total harga tiket, jumlah tiket, dan kode voucher. `int()` digunakan untuk mengubah total harga dan jumlah tiket menjadi integer.

3. Menampilkan data reservasi
```
print(f"Nama Penonton           : {nama_3028}")
print(f"Status Keanggotaan      : {status_member_3028}")
print(f"Total Harga Tiket       : Rp{total_harga_3028}")
print(f"Jumlah Tiket            : {jumlah_tiket_3028}")
print(f"Kode Voucher            : {kode_voucher_3028}")
```
Menampilkan kembali data yang telah dimasukkan. f-string digunakan agar nilai variabel dapat langsung dimasukkan ke dalam teks.

4. Menentukan daftar voucher yang tersedia
`daftar_voucher_3028 = ["NONTON10", "NONTON20", "HEMATWEEKEND", "SPESIALMHS"]`
Membuat sebuah list yang berisi kode-kode voucher yang valid.

5. Mengecek syarat harga
`syarat_harga_3028 = total_harga_3028 >= 150000`
Mengecek apakah total harga minimal Rp150.000. Hasilnya berupa True atau False.

6. Mengecek jumlah tiket
`syarat_jumlah_3028 = jumlah_tiket_3028 >= 2`
Mengecek apakah penonton membeli minimal 2 tiket.

7. Mengecek status member
`status_valid_3028 = status_member_3028 == "member"`
Mengecek apakah status keanggotaan sama dengan "member". Jika sama persis, hasilnya True.

8. Mengecek voucher
`voucher_valid_3028 = kode_voucher_3028 in daftar_voucher_3028`
Operator `in` digunakan untuk mengecek apakah kode voucher yang dimasukkan terdapat di dalam list `daftar_voucher_3028`.

9. Menampilkan hasil validasi
```
print(f"Harga >= Rp150000       : {syarat_harga_3028}")
print(f"Jumlah Tiket >= 2       : {syarat_jumlah_3028}")
print(f"Status Member           : {status_valid_3028}")
print(f"Voucher Tersedia        : {voucher_valid_3028}")
print(f"Mendapatkan Diskon      : {syarat_harga_3028 or syarat_jumlah_3028}")
print(f"Mendapatkan Voucher     : {voucher_valid_3028}")
```
Menampilkan hasil dari semua pengecekan. `or` digunakan agar penonton dapat diskon jika **salah satu** syarat harga atau jumlah tiket terpenuhi.

10. Menghitung diskon dan total pembayaran
```
diskon_3028 = 0.10
print(f"Diskon                  : Rp{diskon_3028 * total_harga_3028}")
print(f"Total Pembayaran        : Rp{total_harga_3028 - total_harga_3028 * diskon_3028}")
print(f"Rata-rata Harga Tiket   : Rp{total_harga_3028 / jumlah_tiket_3028}")
```
Diskon ditetapkan 10% dari total harga. Total pembayaran adalah harga dikurangi diskon, dan rata-rata harga tiket dihitung dengan pembagian.

11. Menampilkan info hak akses
```
print("Kode Hak Akses          : bit0=Member, bit1=Harga, bit2=Jumlah, bit3=Voucher")
print(f"Member Access           : {status_valid_3028}")
print(f"Voucher Access          : {voucher_valid_3028}")
```
Menjelaskan bahwa tiap syarat akan diwakili oleh satu bit (posisi) dalam kode biner.

12. Membentuk kode biner 4-bit
```
kode_reservasi_3028 = int(status_valid_3028) << 0 | int(syarat_harga_3028) << 1 | int(syarat_jumlah_3028) << 2 | int(voucher_valid_3028) << 3
kode_referensi_3028 = int(status_valid_3028) << 0 | int(syarat_harga_3028) << 1 | int(voucher_valid_3028) << 3
```
Setiap status boolean diubah jadi 0/1 (`int(...)`), lalu digeser (`<<`) ke posisi bit-nya masing-masing, kemudian digabung dengan `|` (OR) menjadi satu angka biner 4-bit. `kode_referensi_3028` sengaja tidak menyertakan bit jumlah tiket (bit2), untuk dibandingkan nanti.

13. Menampilkan rincian dan hasil kode biner
```
print(f"{format(int(status_valid_3028) << 0, '04b')} | {format(int(syarat_harga_3028) << 1, '04b')} | ...")
print(f"Kode Biner              : {format(kode_reservasi_3028, '04b')}")
print(f"Kode Desimal            : {kode_reservasi_3028}")
```
`format(x, '04b')` menampilkan angka dalam bentuk biner 4 digit. Baris ini memperlihatkan tiap bit sebelum digabung, lalu hasil gabungannya.

14. Operasi AND — cek status member
```
print(f"Hasil Biner    : {format(kode_reservasi_3028 & (int(status_valid_3028) << 0), '04b')}")
```
Operator `&` (AND) dipakai untuk mengecek apakah bit member (bit0) aktif di `kode_reservasi_3028`. Jika hasilnya bukan 0, berarti status member aktif.

15. Operasi AND — cek voucher
```
print(f"Hasil Biner    : {format(kode_reservasi_3028 & (int(voucher_valid_3028) << 3), '04b')}")
```
Sama seperti sebelumnya, tapi memeriksa bit voucher (bit3).

16. Operasi XOR — bandingkan dua kode
```
print(f"Hasil Biner    : {format(kode_reservasi_3028 ^ kode_referensi_3028, '04b')}")
```
Operator `^` (XOR) membandingkan `kode_reservasi_3028` dan `kode_referensi_3028`. Bit yang berbeda antara keduanya akan bernilai 1, sedangkan bit yang sama bernilai 0 — di sini akan terlihat perbedaan pada bit jumlah tiket (bit2), karena `kode_referensi_3028` tidak menyertakannya.

17. Operasi Shift — upgrade level akses
```
print(f"Hasil Biner    : {format(kode_reservasi_3028 << 1, '04b')}")
```
Operator `<<` menggeser semua bit `kode_reservasi_3028` satu posisi ke kiri, mensimulasikan "kenaikan level akses" (nilainya jadi dua kali lipat).

18. Penutup
`print("\n=== SELESAI ===")`
Menandakan program telah selesai dijalankan.
</div>

---
<div style="height:1123px;font-family: 'Times New Roman', Times, serif;font-size: 18px;">
<h1 align="center">BAB III<br>KESIMPULAN</h1>

Berdasarkan praktikum yang telah dilakukan, dapat disimpulkan bahwa Python menyediakan berbagai jenis operator yang masing-masing memiliki fungsi berbeda, yaitu operator aritmatika untuk perhitungan angka, operator perbandingan dan logika untuk pengambilan keputusan, operator penugasan untuk menyimpan nilai ke variabel, operator bitwise untuk manipulasi data pada level bit, serta operator identitas dan keanggotaan untuk memeriksa objek maupun keberadaan nilai dalam suatu struktur data.

Melalui praktik langsung menuliskan dan menjalankan kode, mahasiswa dapat memahami cara kerja setiap operator serta hasil yang dihasilkannya, sehingga pemahaman yang diperoleh tidak hanya bersifat teoritis, tetapi juga aplikatif. Kemampuan ini menjadi bekal penting untuk mempelajari konsep pemrograman yang lebih kompleks pada tahap selanjutnya, seperti struktur kontrol, fungsi, dan struktur data.
</div>

---
<div style="height:1123px;font-family: 'Times New Roman', Times, serif;font-size: 18px;">
<h1 align="center">DAFTAR PUSTAKA</h1>

1. Python Software Foundation. (2024). The Python Tutorial — An Informal Introduction to Python. Diakses dari https://docs.python.org/3/tutorial/introduction.html

2. Python Software Foundation. (2024). The Python Language Reference — Expressions (Operator Precedence, Comparisons, Boolean Operations). Diakses dari https://docs.python.org/3/reference/expressions.html

3. Python Software Foundation. (2024). The Python Standard Library — Built-in Types (Bitwise Operations on Integer Types). Diakses dari https://docs.python.org/3/library/stdtypes.html#bitwise-operations-on-integer-types

4. Downey, A. (2015). Think Python: How to Think Like a Computer Scientist (2nd ed.). O'Reilly Media.

5. Zelle, J. M. (2016). Python Programming: An Introduction to Computer Science (3rd ed.). Franklin, Beedle & Associates.

6.  Lutz, M. (2013). Learning Python (5th ed.). O'Reilly Media.

</div>